src/LndClient/Data/Newtype.hs

```diff
@@ -111,16 +127,34 @@ newtype NodeLocation = NodeLocation {unNodeLocation :: Text}
 
 instance Out NodeLocation
 
-newtype AddIndex = AddIndex {unAddIndex :: Word64}
-  deriving newtype (PersistField, PersistFieldSql, Eq, Ord, Show)
+newtype AddIndex = AddIndex {unAddIndex :: Natural}
+  deriving newtype (Eq, Ord, Show)
   deriving stock (Generic)
 
```

Comment:
In general - your instances for Persist classes are good and working, but do you see a lot of repeating code? The difference usually is just Word32/Word64 and the type you are converting into. You can create polymorphic functions to reuse in instances. Something like this (if you want) - couple of simple generic combinators, you provide types, accessor and constructor as an arguments:
```haskell
instance PersistField AddIndex where
  toPersistValue = unsafeToPersistValue @Natural @Word64 unAddIndex
  fromPersistValue = safeFromPersistValue @Word64 @Natural AddIndex
```

Reply:
But it's ok to keep it as it is, not that big profit from amount of code